From 82c0cc90905a7ca5984bd35dca7073e583777046 Mon Sep 17 00:00:00 2001
From: Gaggery Tsai <gaggery.tsai@intel.com>
Date: Tue, 28 Apr 2026 15:00:00 -0500
Subject: [PATCH] drm/i915/alpm: limit PR ALPM programming to Panel Replay

PR_ALPM_CTL controls Panel Replay behavior, but the current path also
programs it whenever Adaptive Sync SDP is supported while Panel Replay
is inactive. Limit that register programming to actual Panel Replay
configurations.

Signed-off-by: Gaggery Tsai <gaggery.tsai@intel.com>
---
 drivers/gpu/drm/i915/display/intel_alpm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/display/intel_alpm.c b/drivers/gpu/drm/i915/display/intel_alpm.c
index c6963ea..c38d613 100644
--- a/drivers/gpu/drm/i915/display/intel_alpm.c
+++ b/drivers/gpu/drm/i915/display/intel_alpm.c
@@ -399,7 +399,7 @@ static void lnl_alpm_configure(struct intel_dp *intel_dp,
 			ALPM_CTL_AUX_LESS_SLEEP_HOLD_TIME_50_SYMBOLS |
 			ALPM_CTL_AUX_LESS_WAKE_TIME(crtc_state->alpm_state.aux_less_wake_lines);
 
-		if (intel_dp->as_sdp_supported) {
+		if (intel_dp->as_sdp_supported && crtc_state->has_panel_replay) {
 			u32 pr_alpm_ctl = get_pr_alpm_as_sdp_transmission_time(crtc_state);
 
 			if (crtc_state->link_off_after_as_sdp_when_pr_active)
-- 
2.55.0

